controllers/operatorpolicy_controller.go

```diff
@@ -108,167 +101,300 @@ func (r *OperatorPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+				} else {
+					// The names match, but the specs don't: report NonCompliance
+					err := r.updateStatus(ctx, policy, mismatchCond("OperatorGroup"), mismatchedObj(&opGroup))
+					if err != nil {
+						return fmt.Errorf("error updating the status for an OperatorGroup that does not match: %w", err)
+					}
+
+					if policy.Spec.RemediationAction.IsEnforce() {
+						desiredOpGroup.ResourceVersion = opGroup.GetResourceVersion()
+
+						err := r.Update(ctx, desiredOpGroup)
+						if err != nil {
+							return fmt.Errorf("error updating the OperatorGroup: %w", err)
+						}
+						desiredOpGroup.SetGroupVersionKind(operatorGroupGVK) // Update stripped this information
+
+						// It was updated and should match now, so report Compliance
+						err = r.updateStatus(ctx, policy, updatedCond("OperatorGroup"), updatedObj(desiredOpGroup))
+						if err != nil {
+							return fmt.Errorf("error updating the status after updating the OperatorGroup: %w", err)
+						}
+					}
+				}
+			} else {
+				// There is an OperatorGroup in the namespace that does not match the name of what is in the policy.
+				// Just creating a new one would cause the "TooManyOperatorGroups" failure.
+				// So, just report a NonCompliant status.
+				missing := missingWantedObj(desiredOpGroup)
+				badExisting := mismatchedObj(&opGroup)
+
+				err := r.updateStatus(ctx, policy, mismatchCond("OperatorGroup"), missing, badExisting)
+				if err != nil {
+					return fmt.Errorf("error updating the status for an OperatorGroup with the wrong name: %w", err)
+				}
+			}
+		} else {
+			// There is one operator group in the namespace, but the policy doesn't specify what it should look like.
 
-	// Case 2: Resources exist, but should not exist (i.e. mustnothave or deletion)
-	if exists && !shouldExist {
-		// Future implementation: clean up resources and delete watches if mustnothave, otherwise inform
-		OpLog.Info("The object exists but should not exist")
-	}
+			// FUTURE: check if the one operator group is compatible with the desired subscription.
 
-	// Case 3: Resources do not exist, and should not exist
-	if !exists && !shouldExist {
-		// Future implementation: Possibly emit a success event
-		OpLog.Info("The object does not exist and is compliant with the mustnothave compliance type")
-	}
+			// For an initial implementation, assume if an OperatorGroup already exists, then it's a good one.
+			// Only update the condition if it reflects a NonCompliant state, to prevent repeats.
+			idx, existingCond := policy.Status.GetCondition(opGroupConditionType)
 
-	// Case 4: Resources exist, and should exist (i.e. update)
-	if exists && shouldExist {
-		// Future implementation: Verify the specs of the object matches the one on the cluster
-		OpLog.Info("The object already exists. Checking fields to verify matching specs")
+			if idx == -1 || existingCond.Status == metav1.ConditionFalse {
+				err := r.updateStatus(ctx, policy, matchesCond("OperatorGroup"), matchedObj(&opGroup))
```

Comment:
Would it be possible to include in a message that there is a preexisting operator group that was not verified since it was not specified in the policy?

Reply:
I have created a condition for this case with the message `the policy does not specify an OperatorGroup but one already exists in the namespace - assuming that OperatorGroup is correct`